core/scripts/chaincli/handler/keeper_upkeep_history.go

```diff
@@ -0,0 +1,173 @@
+package handler
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"log"
+	"math/big"
+	"os"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/smartcontractkit/chainlink/core/services/keeper"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/ethkey"
+)
+
+const (
+	defaultMaxBlocksRange = 1000
+)
+
+// UpkeepHistory prints the checkUpkeep status and keeper responsibility for a given upkeep in a set block range
+func (k *Keeper) UpkeepHistory(ctx context.Context, upkeepId int64, from, to uint64) {
```

Comment:
We might need to add another param: `gasPrice`, since some upkeeps now depend on `tx.gasPrice` to make their decisions. I think we can make it optional and default it to 0.